Tests/AppTests/AppTests.swift

```diff
@@ -1,13 +1,258 @@
-import App
+@testable import App
+import Stevenson
+import Vapor
 import XCTest
 
 final class AppTests: XCTestCase {
-    func testNothing() throws {
-        // Add your tests here
-        XCTAssert(true)
+    static let fakeCommits = [
+        "[CNSMR-2044] Commit 1",
+        "[CRP-4141] Commit 3",
+        "[CNSMR-2045] Commit 2",
+        "[CRP-4142] Commit 4"
+    ]
+
+    func testJiraDocumentFromCommits() throws {
+        let release = try GitHubService.Release(
+            repo: GitHubService.Repository(fullName: "company/project", baseBranch: "develop"),
+            branch: "release/app/1.2.3"
+        )
+
+        let entries = ChangelogSection.makeSections(from: AppTests.fakeCommits, for: release)
+        XCTAssertEqual(entries.count, 2)
+        XCTAssertEqual(entries[0].board, "CNSMR")
+        XCTAssertEqual(entries[0].commits.map { $0.message }, ["[CNSMR-2044] Commit 1", "[CNSMR-2045] Commit 2"])
+        XCTAssertEqual(entries[0].commits.map { $0.ticket?.key }, ["CNSMR-2044", "CNSMR-2045"])
+        XCTAssertEqual(entries[1].board, "CRP")
+        XCTAssertEqual(entries[1].commits.map { $0.message }, ["[CRP-4141] Commit 3", "[CRP-4142] Commit 4"])
+        XCTAssertEqual(entries[1].commits.map { $0.ticket?.key }, ["CRP-4141", "CRP-4142"])
+
+        let jiraBaseURL = URL(string: "https://babylonpartners.atlassian.net:443")!
+        let changelogDoc = JiraService.document(from: entries, jiraBaseURL: jiraBaseURL)
+        
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let changelogJsonData = try encoder.encode(changelogDoc)
+        let changelogJson = String(data: changelogJsonData, encoding: .utf8)
+
+        XCTAssertNotNil(changelogJson)
+        #if !os(Linux)
+        add(attachment(name: "Changelog JSON", string: changelogJson))
+        #endif
+
+        let crpConfig = RepoMapping.CRP(
+            environment: .appStore,
+            jiraSummary: { _ in "Fake-Publish Dummy App v1.2.3" }
+        )
+        let issue = JiraService.makeCRPIssue(
+            jiraBaseURL: jiraBaseURL,
+            crpConfig: crpConfig,
+            release: release,
+            changelog: changelogDoc
+        )
+
+        let issueData = try encoder.encode(issue)
+        let issueJson = String(data: issueData, encoding: .utf8)
+
+        #if !os(Linux)
+        add(attachment(name: "Ticket", string: issueJson))
+        #endif
+
+        let expectedTicketJson = #"""
```

Comment:
I thought about it, just not sure what's the best way to load resources compatible with Linux (as there's no bundles on Linux), but will take a look at it 👍

Reply:
@ilyapuchka  Ok this is more complicated that it seems in order to make it work on all environments (Xcode's Cmd-U + `vapor test` on Mac + `vapor test` inside the container). I'm not confident enough in Vapor (especially in how to declare data files to be placed in a specific location in the container when running the tests on CI) yet, so I think I'm gonna keep it as a string literal for now, just extracting that in an `extension` outside of the test's body to avoid polluting the test's code.
So I prefer not to spend too much time on it now, but we could indeed take a look at how to load fixture files in a universal way (Xcode/Mac CLI/CI Container) in a future PR I guess.